### PR TITLE
exporters/otlp/otlptrace: fix incorrect documentation

### PR DESCRIPTION
**Pull request by kevinburkesegment**

### CODE REVIEW

**Review by XSAM — APPROVED** [review]

**Comment by kevinburkesegment** [comment]
Apologies, you may want to do a re-review here, also I am having some difficulty with the `make precommit` target.

```
go: go.opentelemetry.io/otel/internal/tools imports
	github.com/golangci/golangci-lint/cmd/golangci-lint imports
	github.com/golangci/golangci-lint/pkg/commands imports
	github.com/gofrs/flock tested by
	github.com/gofrs/flock.test imports
	gopkg.in/check.v1: go.tmz.dev/musttag@v0.7.2: unrecognized import path "go.tmz.dev/musttag": parse https://go.tmz.dev/musttag?go-get=1: no go-import meta tags ()
	```

File: internal/shared/otlp/otlptrace/otlpconfig/options.go.tmpl [hunk 1]
```diff
@@ -256,13 +256,18 @@ func NewGRPCOption(fn func(cfg Config) Config) GRPCOption {
 
 // Generic Options
 
+// WithEndpoint configures the trace host and port only; endpoint should
+// resemble "example.com" or "localhost:4317". To configure the scheme and path,
+// use WithEndpointURL.
```
Comment (XSAM) [inline internal/shared/otlp/otlptrace/otlpconfig/options.go.tmpl:261]: Could you make this more consistent, like

```suggestion
// WithEndpoint sets the target endpoint (host and port) the Exporter will
// connect to. The provided endpoint should resemble "example.com:4318" (no
// scheme or path).
```

File: internal/shared/otlp/otlptrace/otlpconfig/options.go.tmpl [hunk 2]
```diff
@@ -256,13 +256,18 @@ func NewGRPCOption(fn func(cfg Config) Config) GRPCOption {
...
 func WithEndpoint(endpoint string) GenericOption {
 	return newGenericOption(func(cfg Config) Config {
 		cfg.Traces.Endpoint = endpoint
 		return cfg
 	})
 }
 
+// WithEndpointURL configures the trace scheme, host, port, and path; the
+// provided value should resemble "https://example.com:4318/v1/traces".
```
Comment (XSAM) [inline internal/shared/otlp/otlptrace/otlpconfig/options.go.tmpl:270]: ditto

**Review by hanyuancheung — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
@XSAM Are you OK if I just resolve your comments and merge the PR as it is?

**Review by XSAM — APPROVED** [review]

**Pull request merged**